### PR TITLE
qt: Move qt to qt5.12

### PR DESCRIPTION
**Pull request by gleugner**
This patch is required to meta-boot2qt to QtAS-5.12.1 and meta-qt5 to 5.12.
* higher priority to pelux since b2qt bbappends need to be patched
* enabling of rocko for b2qt
* adding examples for wayland to make debugging easier in neptune-dev image
* Fixes and workarounds to build gstreamer and packagegroup-b2qt-embedded-qt5-toolchain-target

### CODE REVIEW

File: conf/variant/common/bblayers.conf [hunk 1]
```diff
@@ -17,4 +17,9 @@ BBLAYERS ?= "                                          \
   ${BSPDIR}/sources/meta-virtualization                \
 "
 
+# Workaround to build b2qt and qt5.12 with rocko
```
Comment (sashko) [inline conf/variant/common/bblayers.conf]: why do we need this in `master`?

File: conf/variant/rpi-qtauto/local.conf.sample [hunk 2]
```diff
@@ -5,3 +5,8 @@ MACHINE = "raspberrypi3"
 RPI_USE_U_BOOT = "1"
 
 LICENSE_FLAGS_WHITELIST = "commercial_faad2"
+
+
+# Improves build time a lot
+# DISTRO_FEATURES_remove = "webengine"
```
Comment (sashko) [inline conf/variant/rpi-qtauto/local.conf.sample]: why is this here?

File: layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb [hunk 3]
```diff
@@ -15,4 +15,6 @@ IMAGE_INSTALL += "\
 	openssh-sftp-server \
 	packagegroup-bistro-debug-utils \
 	vim \
+    qtdeclarative-tools \
```
Comment (sashko) [inline layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb]: spaces

File: layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb [hunk 4]
```diff
@@ -15,4 +15,6 @@ IMAGE_INSTALL += "\
...
+    qtwayland-examples \
```
Comment (sashko) [inline layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb]: why do we need this?

File: recipes-temporary-patches/gstreamer/gstreamer1.0-plugins-bad_%.bbappend [hunk 5]
```diff
@@ -3,6 +3,10 @@
 # we disable introspection.
 EXTRA_OECONF_aarch64 += "--disable-introspection"
 
+# We overwrite packageconfig since b2qt provides configure options which are
+# not compatible with our version of gstreamer
+PACKAGECONFIG[qt5] = ''
```
Comment (sashko) [inline recipes-temporary-patches/gstreamer/gstreamer1.0-plugins-bad_%.bbappend]: afair, this packageconfig was removed from newer versions

**Comment by sashko** [comment]
@gleugner we need some background story of this PR as personally I am not sure what you're trying to achieve besides trying to make it build on `rocko`, but the question is then: why is this going to `master`?

File: conf/variant/common/bblayers.conf (hunk 1 above)
Comment (thenor) [reply conf/variant/common/bblayers.conf]: I also wonder this, since pelux master is synced with the yocto release thud. Are there parts of boot2qt for Qt 5.12 which are still on rocko, is that why?

Comment (gleugner) [reply conf/variant/common/bblayers.conf]: When building without I get
```
➜  sources bitbake core-image-pelux-qtauto-neptune-dev
ERROR: Unable to start bitbake server (None)
ERROR: Server log for this session (/home/yoctouser/pelux_changes/build/bitbake-cookerdaemon.log):
--- Starting bitbake server pid 5098 at 2019-03-26 14:44:07.877195 ---
Traceback (most recent call last):
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/cookerdata.py", line 284, in parseBaseConfiguration
    self.data = self.parseConfigurationFiles(self.prefiles, self.postfiles)
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/cookerdata.py", line 398, in parseConfigurationFiles
    % (c, " ".join(layerseries), " ".join(compat)))
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/__init__.py", line 120, in fatal
    raise BBHandledException()
bb.BBHandledException

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/daemonize.py", line 83, in createDaemon
    function()
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/server/process.py", line 469, in _startServer
    self.cooker = bb.cooker.BBCooker(self.configuration, self.featureset)
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/cooker.py", line 210, in __init__
    self.initConfigurationData()
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/cooker.py", line 375, in initConfigurationData
    self.databuilder.parseBaseConfiguration()
  File "/home/yoctouser/pelux_changes/sources/poky/bitbake/lib/bb/cookerdata.py", line 317, in parseBaseConfiguration
    raise bb.BBHandledException
bb.BBHandledException
ERROR: Layer b2qt-distro is not compatible with the core layer which only supports these series: thud (layer is compatible with sumo)
```

Comment (gleugner) [reply conf/variant/common/bblayers.conf]: > why do we need this in `master`?

What branch do you suggest?

Comment (sashko) [reply conf/variant/common/bblayers.conf]: what exactly are you building, PELUX? which version of the manifest do you use?

File: conf/variant/rpi-qtauto/local.conf.sample (hunk 2 above)
Comment (gleugner) [reply conf/variant/rpi-qtauto/local.conf.sample]: This reduces the build time in the area of hours. Web engine depends on qt and usually, you don't want to rebuild it, so we should mention this somewhere to save developers time.

File: layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb (hunk 3 above)
Comment (gleugner) [reply layers/b2qt/recipes-core/images/core-image-pelux-qtauto-neptune-dev.bb]: thx

File: conf/variant/rpi-qtauto/local.conf.sample (hunk 2 above)
Comment (sashko) [reply conf/variant/rpi-qtauto/local.conf.sample]: Trust me, we're aware of that :)
